src/scattdata.cpp

```diff
@@ -65,7 +65,10 @@ ScattData::base_combine(size_t max_order,
   xt::xtensor<double, 3> this_matrix({groups, groups, max_order}, 0.);
   xt::xtensor<double, 2> mult_numer({groups, groups}, 0.);
   xt::xtensor<double, 2> mult_denom({groups, groups}, 0.);
-
+  // TODO: Need to review this:
+  if (this->scattxs.size() > 0) {
+    this_matrix = this->get_matrix(max_order);
+  }
```

Comment:
Wouldn't this initialization, with the for loop at lines 97 to 99 double count the first matrix (i.e., the one from those_scatts[0])?

Reply:
As I understand the logic of the code .. it can be written as follows:
`this->scatter_matrix = amin * scattermatrix[temperature_min] ;
 this->scatter_matrix = this->scatter_matrix  + amax*scatter_matrix[temperature_max]` 
At the first iteration of a cycle in xsdata.cpp function `combine` 
At lines 97-99 `scalars` consist from element `{amin}` with `that_scatter` or `those_scatts[0]` is pointed to `scattermatrix[temperature_min]` 
at the second one `scalars` consist from element `{amax}` with `that_scatter` or `those_scatts[0]` is pointed to `scattermatrix[temperature_max]`. And the result we get this_matrix equal to this part `amax*scatter_matrix[temperature_max]`

Reply:
Shortly we didn't save a value of scatter_matrix after first cycle